### DISCUSSION TITLE
Addition of Claim Expired Error Type

### DISCUSSION THREAD

**Discussion by sjclemmy** [panva/jose#62] [Q&A]
I'm using the `JWT.verify` function to determine if a token is expired. The process is:
- call `JWT.verify`
- catch the error
- check that the error code is `ERR_JWT_CLAIM_INVALID` and that the error message is `"exp" claim timestamp check failed`, if so, initiate a token refresh.

It would be nice if I didn't have to rely on the message to know that the token has expired.
What do you think of the addition of a new Error - 'ERR_JWT_CLAIM_EXPIRED'?

I could then just catch the error and check the code rather than the code and the message.

**Comment by panva** [comment]
well, depending on the error message is not a good idea, they get changed overtime and are not part of the semver contract (they are not documented). The error codes on the other hand, are.

I'll think about the right way to move forward.

**Comment by panva** [comment]
@sjclemmy see if https://github.com/panva/jose/compare/jwt-claim-error-improvements helps and come back with feedback please.

**Comment by sjclemmy** [comment]
Thanks @panva - That will definitely serve my purpose. Although it feels a bit non-standard. Wouldn't it be better to just have a new error JWTClaimExpired, rather than JWTClaimInvalid with and additional reason?

**Comment by panva** [comment]
@sjclemmy how about now?

**Comment by sjclemmy** [comment]
Thanks @panva much appreciated :)
